freqtrade/optimize/backtesting.py

```diff
@@ -867,28 +873,76 @@ def run_protections(
-                        # Remove trade due to entry timeout expiration.
-                        return True
-                    else:
-                        # Close additional entry order
-                        del trade.orders[trade.orders.index(order)]
-                if order.side == trade.exit_side:
-                    self.timedout_exit_orders += 1
-                    # Close exit order and retry exiting on next signal.
+    def check_order_cancel(self, trade: LocalTrade, order: Order, current_time) -> bool:
+        """
+        Check if current analyzed order has to be canceled.
+        Returns True if the trade should be Deleted (initial order was canceled).
+        """
+        timedout = self.strategy.ft_check_timed_out(trade, order, current_time)
+        if timedout:
+            if order.side == trade.entry_side:
+                self.timedout_entry_orders += 1
+                if trade.nr_of_successful_entries == 0:
+                    # Remove trade due to entry timeout expiration.
+                    return True
+                else:
+                    # Close additional entry order
                     del trade.orders[trade.orders.index(order)]
+            if order.side == trade.exit_side:
+                self.timedout_exit_orders += 1
+                # Close exit order and retry exiting on next signal.
+                del trade.orders[trade.orders.index(order)]
+
+        return False
 
+    def check_order_replace(self, trade: LocalTrade, order: Order, current_time,
+                            row: Tuple) -> bool:
+        """
+        Check if current analyzed entry order has to be replaced and do so.
+        If user requested cancellation and there are no filled orders in the trade will
+        instruct caller to delete the trade.
+        Returns True if the trade should be deleted.
+        """
+        # only check on new candles for open entry orders
+        if order.side == trade.entry_side and current_time > order.order_date_utc:
+            requested_rate = strategy_safe_wrapper(self.strategy.adjust_entry_price,
+                                                   default_retval=order.price)(
+                trade=trade, order=order, pair=trade.pair, current_time=current_time,
+                proposed_rate=row[OPEN_IDX], current_order_rate=order.price,
+                entry_tag=trade.enter_tag, side=trade.trade_direction
+            )  # default value is current order price
+
+            # cancel existing order whenever a new rate is requested (or None)
+            if requested_rate == order.price:
+                # assumption: there can't be multiple open entry orders at any given time
+                return False
+            else:
+                del trade.orders[trade.orders.index(order)]
+
+            # place new order if None was not returned
+            if requested_rate:
+                self._enter_trade(pair=trade.pair, row=row, trade=trade,
```

Comment:
I'm not entirely sure this call is correct.
We expect to update an existing trade - but this is creating a new trade object (however not properly storing that).

The test i'v added shows (if you debug it) that the open_rate is left at the initial rate - so there is clearly something fishy going on here.

Reply:
Proposed solution for this added through https://github.com/freqtrade/freqtrade/pull/6692/commits/dbecc097dffdb54b8b78c10ce52c99283c00d4b0

Reply:
Personally I'm not particularly happy with needing this:
`latest_order_in_trade.id is not None):`
I'm assuming is irrelevant for normal operation but for testcases it seems that orders being injected, they might not have an `id` populated.

Reply:
I fear i did misslead you (or rather - didn't analyze properly before posting the above).
`open_rate` is properly set in `update_trade()` - which covers the live scenario.

This method is however not called in backtesting - so the setting of open rate should be moved to backtesting - where the (first) order fills.

Reply:
Yes somehow I've missed the failing freqtradebot tests, or ommitted to run them. Would have been maybe obvious to me also that it would have negative impact in live/dry side.
Another swing at this added through: https://github.com/freqtrade/freqtrade/pull/6692/commits/2bed0eab0cc5433800ac8de8b68bb0ad2d7ca1f3